lib/include/public/PauseManager.hpp

```diff
@@ -0,0 +1,48 @@
+//
+// Copyright (c) 2015-2020 Microsoft Corporation and Contributors.
+// SPDX-License-Identifier: Apache-2.0
+//
+#ifndef PAUSEMANAGER_HPP
+#define PAUSEMANAGER_HPP
+
+#include <atomic>
+#include <chrono>
+#include <condition_variable>
+#include <mutex>
+
+#include "ctmacros.hpp"
+
+namespace MAT_NS_BEGIN
+{
+    class PauseManager {
+        public:
+
+        static void PauseActivity() noexcept;
+        static void ResumeActivity() noexcept;
+        static void QuiesceWait() noexcept;
+        static void QuiesceWaitFor(std::chrono::milliseconds timeout) noexcept;
+        static bool TryLock() noexcept;
+        static void ReleaseLock() noexcept;
+
+        class Lock {
```

Comment:
I'm wondering what value-add is to introduce a separate `Lock` - we can probably reuse the existing `LOCKGUARD` macro instead?